wps_tools/utils.py

```diff
@@ -113,13 +114,18 @@ def build_meta_link(
     return meta_link.xml
 
 
-def log_handler(process, response, message, process_step=None, level="INFO"):
+def log_file_path(process):  # From Finch bird
+    """Returns the filepath to write the process logfile."""
+    return Path(process.workdir) / "log.txt"
+
+
+def log_handler(process, response, message, logger, process_step=None):
     if process_step:
         status_percentage = process.status_percentage_steps[process_step]
     else:
         status_percentage = response.status_percentage
 
     # Log to all sources
-    pywps_logger.log(getattr(logging, level), message)
-    stderr_logger.log(getattr(logging, level), message)
+    logger.log(logger.level, message)
+    log_file_path(process).open("a", encoding="utf8").write(message + "\n")
     response.update_status(message, status_percentage=status_percentage)
```

Comment:
Urgh this method is proving to be a bit tricky to get right. I was wrong about being able to log directly with the logger's level. The level may be defined as `INFO` but a user may want to log at a different level depending on the message So we'd have to have a `level` param.  Furthermore, we should probably allow the user to name the log file if they so choose and just leave `"log.txt"` as the default name.  That would bring our number of parameters up to an uncomfortable level.

To reduce some of these params we have to combine a couple items.
1. Let's go back to just passing in a `log_level` and using a logger that is defined here. This removes the need for both `logger` and `log_level`.
2. We can combine the `process` and `process_step` params. This is a bit ugly so if you have suggestions please let me know. Basically we just want to take in an optional percentage and if its not there we use `response.status_percentage`.

Try adding those features for now, and we can come back to it.

Reply:
The first combination is fine, but the second one may not be feasible because `log_file_path` looks in `process.workdir`, so that parameter's necessary. I'll see what we can combine.

Reply:
Unfortunately, other than using `log_level` instead of `logger`, I haven't figured out a way to reduce the number of parameters in a clean way. Is six too many?